src/main/java/org/mifos/connector/mpesa/utility/MpesaUtils.java

```diff
@@ -31,6 +32,31 @@ public class MpesaUtils {
     @Value("${roster.host}")
     private static String rosterHost;
 
+    public static ChannelSettlementRequestDTO convertPaybillToChannelPayload(PaybillRequestDTO paybillConfirmationRequestDTO, String amsName) {
+        JSONObject payer = new JSONObject();
+        JSONObject partyIdInfoPayer = new JSONObject();
+        partyIdInfoPayer.put("partyIdType", "MSISDN");
+        partyIdInfoPayer.put("partyIdentifier", paybillConfirmationRequestDTO.getMsisdn());
+        payer.put("partyIdInfo", partyIdInfoPayer);
+
+        JSONObject payee = new JSONObject();
+        JSONObject partyIdInfoPayee = new JSONObject();
+        if (amsName.equalsIgnoreCase("paygops")) {
+            partyIdInfoPayee.put("partyIdType", "FOUNDATIONALID");
+            partyIdInfoPayee.put("partyIdentifier", paybillConfirmationRequestDTO.getBillRefNo());
+        } else if (amsName.equalsIgnoreCase("roster")) {
+            partyIdInfoPayee.put("partyIdType", "ACCOUNTID");
+            partyIdInfoPayee.put("partyIdentifier", paybillConfirmationRequestDTO.getBillRefNo());
+        }
+        payee.put("partyIdInfo", partyIdInfoPayee);
+
+        JSONObject amount = new JSONObject();
+        amount.put("amount", paybillConfirmationRequestDTO.getTransactionAmount());
+        amount.put("currency", "TZS");
```

Comment:
Currency should not be hardcoded.

Reply:
we don't receive currency as part of C2B mpesa payload.

Reply:
can we populate from the webhook url ?

Reply:
added it from params